### PR TITLE
Backport 481

### LINKED ISSUE #461: crazy contours when switching between no smooth and box smooth
Seen in the mac v1.3beta and dev on mix n match

To reproduce:
1) load an image
2) create contours with "no smooth" mode
3) switch to "box" smooth and regenerate contours 

M100_combine_CO_cube image mom0 pbcor-image-2020-3-6-14-48-18

test image attached
[M100_12m_CO_central.image.mom0.fits.zip](https://github.com/CARTAvis/carta-frontend/files/4296898/M100_12m_CO_central.image.mom0.fits.zip)

### PR DESCRIPTION
**Pull request by veggiesaurus**
Backport to fix #461

### CODE REVIEW

**Comment by veggiesaurus** [comment]
@kswang1029 please check that images with and without NaNs are behaving correctly to ensure no regressions

**Comment by confluence** [comment]
Did you mean to make this PR against the release?

File: ImageData/CartaHdf5Image.cc [hunk 1]
```diff
@@ -21,8 +21,8 @@ CartaHdf5Image::CartaHdf5Image(
     : casacore::ImageInterface<float>(casacore::RegionHandlerHDF5(GetHdf5File, this)),
       _valid(false),
       _pixel_mask(nullptr),
-      _mask_spec(mask_spec) {
-    _lattice = casacore::HDF5Lattice<float>(casacore::CountedPtr<casacore::HDF5File>(new casacore::HDF5File(filename)), array_name, hdu);
```
Comment (confluence) [inline ImageData/CartaHdf5Image.cc:25]: I'm confused by this diff -- this change should already have been merged into both dev and the release.

File: ImageData/FileLoader.cc [hunk 2]
```diff
@@ -94,11 +94,11 @@ bool FileLoader::FindCoordinateAxes(IPos& shape, int& spectral_axis, int& stokes
     }
 
     // use CoordinateSystem to find coordinate axes
-    casacore::Vector<casacore::Int> linear_axes = coord_sys.linearAxesNumbers();
```
Comment (confluence) [inline ImageData/FileLoader.cc:97]: Is this a side effect?

File: FileList/FileExtInfoLoader.cc [hunk 3]
```diff
@@ -237,10 +237,11 @@ bool FileExtInfoLoader::FillFileInfoFromImage(CARTA::FileInfoExtended* extended_
                 }
 
                 int spectral_axis, stokes_axis;
-                _loader->FindCoordinateAxes(image_shape, spectral_axis, stokes_axis, message);
-                AddShapeEntries(extended_info, image_shape, spectral_axis, stokes_axis);
-                AddComputedEntries(extended_info, image, radesys);
-                file_ok = true;
+                if (_loader->FindCoordinateAxes(image_shape, spectral_axis, stokes_axis, message)) {
```
Comment (confluence) [inline FileList/FileExtInfoLoader.cc:240]: Is this a side effect?

**Review by confluence — COMMENTED** [review]
Should this be a PR against the release branch? Either way, there seem to be some unrelated changes.

**Comment by veggiesaurus** [comment]
sorry, as @confluence mentioned, this should be a PR against `release/1.3` :laughing:

**Pull request closed without merging**